bpaotu/bpaotu/views.py

```diff
@@ -495,6 +495,226 @@ def str_none_blank(v):
         return csv_fd.getvalue()
 
 
+
+
+@require_http_methods(["GET"])
+def otu_biom_export(request):
+    SAMPLE_RUN_VAL = 1
+    TRIAL_SWITCH = False
+
+
+
+    def val_or_empty(obj):
+        if obj is None:
+            return ''
+        return obj.value
+
+    def write_output(val, stream, file_ptr):
+        stream.write(val)
+        file_ptr.write(stream.getvalue())
+        stream.seek(0)
+        stream.truncate(0)
+
+    with open('json_biomfile.biom', 'w') as biom_fp:
+        # Write out the JSON file header first
+        fd = StringIO()
+
+        biom_file = OrderedDict()
+
+        biom_file['id'] = None
+        biom_file['format'] = "1.0.0"
+        biom_file['format_url'] = "http://biom-format.org"
+        biom_file['type'] = "OTU table"
+        biom_file['generated_by'] = "Bioplatforms Australia"
+        biom_file['date'] = datetime.datetime.now().replace(microsecond=0).isoformat()
+        biom_file['matrix_type'] = "sparse"
+        biom_file['matrix_element_type'] = "int"
+
+        biom_header = "{"
+
+        for key, val in biom_file.items():
+            biom_header += '"{}": "{}",\n'.format(key, val)
+
+        write_output(biom_header, fd, biom_fp)
+
+        rows = []
+        columns = []
+
+        params, errors = param_to_filters(request.GET['q'])
+        with SampleQuery(params) as query:
+
+
+
+            logger.critical("--------------------------------------------------------------------------------")
+
+
+
+            # Write out the OTU list (which form the rows)
+            q = query.matching_otus()
+            result_length = q.distinct('code').count()
+            logger.critical(">>>>>>>>>>>>>>>>>>>>>>>>>{}".format(result_length))
+            if TRIAL_SWITCH:
+                result_length = SAMPLE_RUN_VAL
+
```

Comment:
I don't know what this TRIAL_SWITCH thing is - if it's just debugging code, remove it.

Reply:
Fixed.